clients/src/test/java/org/apache/kafka/clients/admin/NewTopicTest.java

```diff
@@ -0,0 +1,168 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class NewTopicTest {
+
+    public static final String TEST_TOPIC = "testtopic";
+    public static final int NUM_PARTITIONS = 3;
+    public static final short REPLICATION_FACTOR = 1;
+    public static final String CLEANUP_POLICY_CONFIG_KEY = "cleanup.policy";
+    public static final String CLEANUP_POLICY_CONFIG_VALUE = "compact";
+    public static final List<Integer> BROKER_IDS = Arrays.asList(1, 2);
+
+    @Test
+    public void testConstructorWithPartitionsAndReplicationFactor() {
+        NewTopic topic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithOptionalValues() {
+        Optional<Integer> numPartitions = Optional.empty();
+        Optional<Short> replicationFactor = Optional.empty();
+        NewTopic topic = new NewTopic(TEST_TOPIC, numPartitions, replicationFactor);
+        assertEquals(TEST_TOPIC, topic.name());
+        assertEquals(NO_NUM_PARTITIONS, topic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, topic.replicationFactor());
+        assertNull(topic.replicasAssignments());
+    }
+
+    @Test
+    public void testConstructorWithReplicasAssignments() {
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(0, BROKER_IDS);
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        assertEquals(TEST_TOPIC, newTopic.name());
+        assertEquals(NO_NUM_PARTITIONS, newTopic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, newTopic.replicationFactor());
+        assertEquals(replicasAssignments, newTopic.replicasAssignments());
+    }
+
+    @Test
+    public void testConfigsNotNull() {
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+        Map<String, String> configs = new HashMap<>();
+        configs.put(CLEANUP_POLICY_CONFIG_KEY, CLEANUP_POLICY_CONFIG_VALUE);
+        newTopic.configs(configs);
+        assertEquals(configs, newTopic.configs());
+    }
+
+    @Test
+    public void testConfigsNull() {
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+        assertNull(newTopic.configs());
+    }
+
+    @Test
+    public void testUnmodifiableReplicasAssignments() {
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(0, BROKER_IDS);
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        Map<Integer, List<Integer>> returnedAssignments = newTopic.replicasAssignments();
+
+        assertThrows(UnsupportedOperationException.class, () -> {
```

Comment:
it seems this can be simplified to `assertThrows(UnsupportedOperationException.class, () -> returnedAssignments.put(1, Arrays.asList(3, 4)));`

Reply:
Done.